pull_request: add button to submit try job (bug 2003366)

zzzeid opened the pull request:
- add button to trigger try job
- add link to treeherder for last try job
- fix bug with response code in try job endpoint
- update admin to show handover job info

shtrom commented on src/lando/ui/pull_requests.py:
```diff
@@ -20,6 +25,14 @@ def get(
+            | Q(name__startswith="infra-testing-")
+            | Q(name__startswith="ff-test-")
+            | Q(name="git-repo")
+        )
```
nit: can we have this at the module level?

shtrom reviewed (approved):
(Preemptive) LGTM.

zzzeid committed:
pull_request: add button to submit try job (bug 2003366)

- add button to trigger try job
- add link to treeherder for last try job
- fix bug with response code in try job endpoint
- update admin to show handover job info

zzzeid committed:
code review feedback
